verification/src/solidity/verifier.rs

```diff
@@ -1,51 +1,281 @@
 #![allow(dead_code, unused)]
 
+use crate::types::Mismatch;
+use bytes::Buf;
+use ethers_core::types::Bytes as DisplayBytes;
 use ethers_solc::CompilerOutput;
-use std::str::FromStr;
+use minicbor::{data::Type, Decode, Decoder};
+use std::{error::Error, str::FromStr};
 use thiserror::Error;
 
 /// Errors that may occur during initial [`Verifier`] setup
 /// with input data provided by the requester.
-#[derive(Clone, Debug, Error)]
-pub(crate) enum InitializationError {}
+#[derive(Clone, Debug, PartialEq, Error)]
+pub(crate) enum InitializationError {
+    #[error("creation transaction input is not a valid hex string")]
+    InvalidCreationTxInput,
+    #[error("deployed bytecode is not a valid hex string")]
+    InvalidDeployedBytecode,
+    #[error("cannot parse metadata hash from deployed bytecode: {0}")]
+    MetadataHashParseError(String),
```

Comment:
I suggest remove "error" suffixes from error enum variants, because it will be mostly used with the enum itself, which already has error in the name, i.e:

```rust
enum FooError {
  BarError,
  Baz,
}

FooError::BarError // two errors, too much
FooError::Baz // one error, nice
```

So I suggest changing `MetadataHashParseError` to `MetadataHashParse`

Reply:
Agree